### ISSUE TITLE
All poppers are broken in fullscreen mode when `zIndex` is modified via `muiTablePaperProps`

### ISSUE THREAD

**Issue by raduciocan** [KevinVandy/material-react-table#923]
### material-react-table version

v2.5.0

### react & react-dom versions

v18.2.0

### Describe the bug and the steps to reproduce it

When specifying a custom `zIndex` inside `muiTablePaperProps`, poppers (such as show/hide columns menu, filter menus in popover filtering mode, etc.) are not rendered correctly when in fullscreen table mode. They get rendered under the fullscreen table, and in their initial non-fullscreen location.

### Minimal, Reproducible Example - (Optional, but Recommended)

https://codesandbox.io/p/devbox/nice-kalam-3pcwyg

### Screenshots or Videos (Optional)

https://github.com/KevinVandy/material-react-table/assets/93911989/319d572c-c159-456a-87cf-dd6593d1535c

### Do you intend to try to help solve this bug with your own PR?

No, because I do not have time to dig into it

### Terms

- [X] I understand that if my bug cannot be reliably reproduced in a debuggable environment, it will probably not be fixed and this issue may even be closed.

**Comment by KevinVandy** [comment]
How big is your custom zIndex? Larger than 200?

**Comment by raduciocan** [comment]
Yes, I had to set it to 1201 since this is the only option that goes above my website toolbar. I am using MUI AppBar component with a Toolbar and a Sidebar too, so I guess that 1200 is the default zIndex of AppBar. I tried tweaking the zIndex of the MUI navigation components but it breaks their behaviour.

UPDATE: It seems to work with 1201, last time I tested and was broken  was with a zIndex of 2000.
